### PR TITLE
Migrate reducers from flow -> ts

### PR DESCRIPTION
**Pull request by Fausto95**

### CODE REVIEW

File: packages/@coorpacademy-progression-engine/src/reducers/content.ts [hunk 1]
```diff
@@ -9,23 +7,26 @@ import type {
...
         return answerAction.payload.content;
       }
       case 'extraLifeAccepted': {
-        const acceptAction = (action: ExtraLifeAcceptedAction);
+        const acceptAction: ExtraLifeAcceptedAction = action;
         return acceptAction.payload.content;
       }
       case 'extraLifeRefused': {
-        const refuseAction = (action: ExtraLifeRefusedAction);
+        const refuseAction: ExtraLifeRefusedAction = action;
```
Comment (VincentCATILLON) [inline packages/@coorpacademy-progression-engine/src/reducers/content.ts]: use enum for the cases please

File: packages/@coorpacademy-progression-engine/src/reducers/is-correct.ts [hunk 2]
```diff
@@ -0,0 +1,14 @@
+import {Action, AnswerAction, Config} from '../types';
+
+export default function isCorrect(config: Config) {
+  return (state: boolean | null = true, action: Action): boolean | null => {
+    switch (action.type) {
+      case 'answer': {
```
Comment (VincentCATILLON) [inline packages/@coorpacademy-progression-engine/src/reducers/is-correct.ts]: use enum

File: packages/@coorpacademy-progression-engine/src/reducers/lives-disabled.ts [hunk 3]
```diff
@@ -0,0 +1,7 @@
+import {Action, Config} from '../types';
+
+export default function livesDisabled(config: Config) {
+  return (state: boolean | void , action: Action): boolean => {
```
Comment (VincentCATILLON) [inline packages/@coorpacademy-progression-engine/src/reducers/lives-disabled.ts]: boolean | null

File: packages/@coorpacademy-progression-engine/src/reducers/remaining-life-requests.ts [hunk 4]
```diff
@@ -1,8 +1,6 @@
-// @flow
+import {Action, Config, State} from '../types';
 
-import type {Action, Config, State} from '../types';
-
-export default function remainingLifeRequests(config: Config): (number, Action, State) => number {
+export default function remainingLifeRequests(config: Config) {
```
Comment (VincentCATILLON) [inline packages/@coorpacademy-progression-engine/src/reducers/remaining-life-requests.ts]: can be an arrow function

File: packages/@coorpacademy-progression-engine/src/reducers/requested-clues.ts [hunk 5]
```diff
@@ -1,14 +1,12 @@
...
-import type {Action, AskClueAction, Config} from '../types';
+import {Action, AskClueAction, Config} from '../types';
 
-export default function requestedClues(config: Config): (Array<string>, Action) => Array<string> {
+export default function requestedClues(config: Config) {
   return (array: Array<string> = [], action: Action): Array<string> => {
     switch (action.type) {
       case 'clue': {
-        const requestedClueAction = (action: AskClueAction);
+        const requestedClueAction = action as AskClueAction;
```
Comment (VincentCATILLON) [inline packages/@coorpacademy-progression-engine/src/reducers/requested-clues.ts]: imo typing constant should be like: `const xxx: Y = z`

File: packages/@coorpacademy-progression-engine/src/reducers/test/fixtures/engines.ts [hunk 6]
```diff
@@ -0,0 +1,14 @@
+export interface FakeContent {
+  ref: 'microlearning' | 'learner',
```
Comment (VincentCATILLON) [inline packages/@coorpacademy-progression-engine/src/reducers/test/fixtures/engines.ts]: can use enum

File: packages/@coorpacademy-progression-engine/src/types.ts [hunk 7]
```diff
@@ -82,15 +82,15 @@ export interface ResourceContent {
 export type Content = GenericContent | ResourceContent;
 
 export interface ViewedResource {
-  type: $PropertyType<Content, 'type'>,
+  type: Pick<Content, 'type'>,
```
Comment (VincentCATILLON) [inline packages/@coorpacademy-progression-engine/src/types.ts]: Content['type']

File: packages/@coorpacademy-progression-engine/src/types.ts [hunk 8]
```diff
@@ -82,15 +82,15 @@ export interface ResourceContent {
...
   ref: string,
   resources: Array<string>
 };
 
 export interface AnswerRecord {
   slideRef: string,
   answer: Answer,
-  isCorrect: ?boolean
+  isCorrect: boolean | void
```
Comment (VincentCATILLON) [inline packages/@coorpacademy-progression-engine/src/types.ts]: incorrect, should be: `isCorrect?: boolean | null`

File: packages/@coorpacademy-progression-engine/src/types.ts [hunk 9]
```diff
@@ -105,7 +105,7 @@ export interface GenericState  {
   chapters?: Array<string>,
   lives: number,
   livesDisabled?: boolean,
-  isCorrect: ?boolean,
+  isCorrect: boolean | void,
```
Comment (VincentCATILLON) [inline packages/@coorpacademy-progression-engine/src/types.ts]: same as above

File: packages/@coorpacademy-progression-engine/src/types.ts [hunk 10]
```diff
@@ -137,8 +137,8 @@ export interface AnswerAction extends GenericAction {
     content: Content,
     nextContent: Content,
     answer: Answer,
-    isCorrect: ?boolean,
-    godMode: ?boolean,
+    isCorrect: boolean | void,
+    godMode: boolean | void,
```
Comment (VincentCATILLON) [inline packages/@coorpacademy-progression-engine/src/types.ts]: same as above

File: packages/@coorpacademy-progression-engine/src/types.ts [hunk 11]
```diff
@@ -236,7 +236,7 @@ export interface Progression {
 
 export interface PartialCorrection {
   answer: string | void,
-  isCorrect: ?boolean
+  isCorrect: boolean | void
```
Comment (VincentCATILLON) [inline packages/@coorpacademy-progression-engine/src/types.ts]: same as above

File: packages/@coorpacademy-progression-engine/src/types.ts [hunk 12]
```diff
@@ -383,7 +383,7 @@ export interface Slide {
   chapter_id: string,
   // $FlowFixMe union type  :(
   question: Question,
-  position?: ?number,
+  position?: number | void,
```
Comment (VincentCATILLON) [inline packages/@coorpacademy-progression-engine/src/types.ts]: same as above

File: packages/@coorpacademy-progression-engine/src/types.ts [hunk 13]
```diff
@@ -330,7 +330,7 @@ export interface TemplateQuestion {
   type: 'template',
   content: {
     matchOrder: boolean,
-    maxTypos?: ?number,
+    maxTypos?: number | void,
```
Comment (VincentCATILLON) [inline packages/@coorpacademy-progression-engine/src/types.ts]: same as above

File: packages/@coorpacademy-progression-engine/src/types.ts [hunk 14]
```diff
@@ -321,7 +321,7 @@ export interface QCMDragQuestion extends QuestionCommon {
 export interface BasicQuestion extends QuestionCommon {
   type: 'basic',
   content: {
-    maxTypos?: ?number,
+    maxTypos?: number | void,
```
Comment (VincentCATILLON) [inline packages/@coorpacademy-progression-engine/src/types.ts]: same as above

File: packages/@coorpacademy-progression-engine/src/types.ts (hunk 8 above)
Comment (godu) [reply packages/@coorpacademy-progression-engine/src/types.ts]: non, c'est bien void

**Pull request merged**